### DISCUSSION TITLE
Alarm clock support

### DISCUSSION THREAD

**Discussion by Nezz** [home-assistant/architecture#1089] [Entity Models]
# Context
I started implementing alarm clock support in https://github.com/home-assistant/core/pull/117755, but I was guided here to propose and have a discussion over it.

Currently, setting up an alarm clock in Home Assistant requires multiple entities, which are cumbersome and inconsistent:
- Switch entity for turning on/off the alarm
- Time entity for the alarm's time
- Select entity for choosing between days. However, this only supports one selection. Therefore a text entity might be used but is really clumsy.

Example from https://github.com/theneweinstein/somneo:
<img width="308" alt="Screenshot 2024-05-20 at 10 56 04" src="https://github.com/home-assistant/architecture/assets/431167/38851784-8fdc-46e1-ba49-591e4c44b2ce">

Due to this, they are often not implemented (see https://github.com/home-assistant/architecture/discussions/1046 and https://github.com/lukas-clarke/eight_sleep/issues/19) even though the functionality is similar to the existing [timer](https://www.home-assistant.io/integrations/timer) integration.

# Objectives

- Simplify the process of setting and managing alarms in Home Assistant.
- Provide a unified entity and user interface for alarm clock functionality.
- Enable seamless integration with existing and future alarm-related integrations, including voice.

# Components

### Unified Alarm Clock Entity:
- Allows scheduling alarms for a certain time of day
- Allows repeating alarms on certain days, or one-off alarms 
- Can be turned on and off
- Fire events, which allows automatons can listen to the alarm going off and turn on lights, open covers, etc.

<img width="739" alt="Screenshot 2024-05-20 at 13 08 27" src="https://github.com/home-assistant/architecture/assets/431167/f7e991a0-e77a-4f26-b6ec-686e24691649">

I did not include any snooze functionality in this because it requires interaction, therefore it's probably best implemented by the actual integration and left out of the entity model.

### Frontend
- Ability to set and modify alarm time.
- Toggle alarm on/off.
- Select multiple days for the alarm (e.g., weekdays, weekends).
- Ability to create alarms as helpers.

**Comment by parautenbach** [comment]
May I add? 

The ability to skip the next alarm (followed by reverting to the existing schedule after that).

**Comment by Nezz** [comment]
That's a good idea. Philips does not support it, but Eight Sleep does. They have implemented it in a way that when you change or turn off the alarm, it asks if you want to do it for tomorrow only or change the routine. Not sure how they implemented it under the hood though.

**Reply by parautenbach** [reply]
Nice. I was going on what my iOS alarm clock does, which I replicated in my HA setup. In my larger setup (just out of interest) I look at workdays and can also add a vacation entry to a calendar to automatically disable the morning alarm (and accompanying morning routine), but sometimes I/we simply want to skip the next one (so that's the context).

**Comment by WinSCaP** [comment]
Sad to see this is not yet implemented. An easy to use alarmclock build into Home Assistant is what a lot of people would want (see the number of blog, forumposts, etc about alarm clocks. I have an alarmclock for our bedroom, now my wife wants me to implement that for the kids as well, that means at least 14 new helpers and multiple automations. This would be a quality of life improvement.

**Comment by EuleMitKeule** [comment]
Having an alarm entity platform would also solve [this problem](https://github.com/home-assistant/frontend/pull/10006) since the alarm would not indicate an unknown state when it is not set.

**Comment by SoCuul** [comment]
This would be extremely useful and would be an incredible QOL improvement for people trying to use home assistant as an alarm clock without tedious work setting up the infrastructure to do so.

I'm surprised this architecture request has not been approved by now.

**Comment by xiasi0** [comment]
It is easy to set a single alarm and turn off the alarm using LLM, but it is not known how to set a recurring alarm based on schedule, working days, and holidays.

**Comment by peteS-UK** [comment]
We've been trying to add alarm support to the squeezebox integration, https://github.com/home-assistant/core/pull/141055, but the current approach which doesn't allow the use of attributes anymore and instead wants multiple entities per alarm, which are only associated together by their name is, IMHO, completely unworkable.  As has been described above, we need an alarm entity, with the associated properties so that we can add alarm support for more platforms in a consistent and user friendly way.

**Comment by kmanc** [comment]
What would it take for this proposal to take hold? It seems reasonable, and something that many would benefit from.

**Comment by chemelli74** [comment]
I need a solution also for the new Amazon Devices integration.
there we have:

- timers
- reminders
- alarms

**Comment by dcapslock** [comment]
Looking at this proposal I am thinking that the current in built schedule would be the best base. Moreover, the addition of improved ways to edit a schedule along with some blueprints for use cases (Alarm 'Clock'). This approach leads to great flexibility while providing for all cases mentioned here.

**Comment by Nezz** [comment]
2025.8 adds weekdays to the time trigger, so there is a bit of progress being made:
https://github.com/orgs/home-assistant/discussions/26
